PythonRegression/tests/features/steps/api_test_steps.py

```diff
@@ -68,377 +76,140 @@ def threaded_call(step,apiCall,node):
     api = api_utils.prepare_api_call(node)
 
     def make_call(node, arg_list):
-        response = api_utils.fetch_call(apiCall, arg_list['api'], arg_list['options'])
-        arg_list['responses'][apiCall] = {}
-        arg_list['responses'][apiCall][node] = response
+        response = api_utils.fetch_call(api_call, arg_list['api'], arg_list['options'])
+        arg_list['responses'][api_call] = {}
+        arg_list['responses'][api_call][node] = response
         return response
 
-    args = {node: {'api': api,'options': options,'responses': world.responses}}
-    future_results = pool.start_pool(make_call,1,args)
+    args = {node: {'api': api, 'options': options, 'responses': world.responses}}
+    future_results = pool.start_pool(make_call, 1, args)
 
     if 'future_results' not in world.config:
         world.config['future_results'] = {}
-    world.config['future_results'][apiCall] = future_results
+    world.config['future_results'][api_call] = future_results
 
     
 @step(r'we wait "(\d+)" second/seconds')
-def wait_for_step(step,time):
+def wait_for_step(step, time):
+    """
+    Wait a specified number of seconds before continuing.
+
+    :param time: The number of seconds you would like the step to wait for.
+    """
     logger.info('Waiting for {} seconds'.format(time))
     sleep(int(time))
 
 
 @step(r'the "([^"]+)" parallel call should return with:')
-def compare_thread_return(step,apiCall):
-    """Prepare response list for comparison"""
+def compare_thread_return(step, api_call):
+    """
+    Prepare response list for comparison.
+
+    :param api_call: The API call you would like to find a response for
+    :param step.hashes: A gherkin table present in the feature file specifying the
+                        values and the associated type to be found in the response.
+    """
     logger.debug(world.responses)
-    future_results = world.config['future_results'][apiCall]
+    future_results = world.config['future_results'][api_call]
 
     for result in future_results:
-        response_list = pool.fetch_results(result,1)
+        response_list = pool.fetch_results(result, 5)
```

Comment:
explain to me the number 5 please?

Reply:
Considering how quickly the code switches from making the call to fetching the result, sometimes 1 second timeouts aren't quite long enough for the return to be present in time, so I increased the timeout to 5 seconds in case one or two calls hang mid call.

Reply:
Should there be a timeout are we supposed to fail?

Reply:
I suppose to say it fails here is insincere. The call will return an empty response if it fails the timeout and during the response check, it will fail because the response fetched is `null`. The previous alternative was that the call would hang and the tests never proceed from that point. (This is assuming we call it non parallel)

Reply:
So the timeout just makes sure that no individual call stops the rest of the tests from running.